### PR TITLE
CircleCI updates, removal of github actions

### LINKED ISSUE #467: Circle CI Improvements
**Task summary**

Circle Improvements: 

- [ ] CodeCov coverage to be added
- [ ] Build specific checks
- [ ] Nightly checks
- [ ] Optimizations
- [ ] Move doc builds from github actions to CircleCI

### PR DESCRIPTION
**Pull request by dutterbutter**
**Summary of changes**
Changes introduced in this pull request:
- Removes github actions
- Moves doc and audit checks to circleci
- Adds macos, nightly, and install build checks
- Adds code coverage using grcov
- Adds codecov badge to repo
- Adds daily build check to main

I think we are able to take advantage of parallelism within container for test vectors as defined [here](https://circleci.com/docs/2.0/parallelism-faster-jobs/) but I could not get it to split the tests out correctly. 

**Reference issue to close (if applicable)**
Closes #467 

**Other information and links**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -140,15 +211,89 @@ jobs:
       - run:
           name: Run test vectors
           command: make run-vectors
-  install:
+  coverage:
+    executor: mac-executor
```
Comment (austinabell) [inline .circleci/config.yml:215]: this should maybe use the linux executor? or is this intentional?

Comment (dutterbutter) [reply .circleci/config.yml:215]: This is intentional, I kept getting unexpected dependency issues while running in the linux executor. I plan to investigate this more but for now should be sufficient.

File: .circleci/config.yml [hunk 2]
```diff
@@ -140,15 +211,89 @@ jobs:
...
+                  export RUSTFLAGS="-Zprofile -Ccodegen-units=1 -Coverflow-checks=off"
+                  make build
+                  git submodule update --init
+                  sudo make run-vectors
+                  sudo make test-all
+            - run:
+                name: Upload to codecov
+                command: |
+                  zip -0 ccov.zip `find . \( -name "forest*.gc*" \) -print`;
+                  ./grcov ccov.zip -s . -t lcov --llvm --branch --ignore-not-existing --ignore "/*" --ignore="target/debug/build/*" --ignore="tests/*" --ignore="blockchain/beacon/tests/*" -o lcov.info;
```
Comment (austinabell) [inline .circleci/config.yml:262]:
```suggestion
                  ./grcov ccov.zip -s . -t lcov --llvm --branch --ignore-not-existing --ignore "/*" --ignore="target/debug/build/*" --ignore="**/tests/*" -o lcov.info;
```

File: .circleci/config.yml (hunk 1 above)
Comment (austinabell) [reply .circleci/config.yml:215]: Maybe add a TODO comment to indicate this, so it doesn't get lost in the future? Or some mechanism to come back to this?

**Review by austinabell — APPROVED** [review]
Awesome work dude :)

**Review by ec2 — APPROVED** [review]

**Comment by dutterbutter** [comment]
@ec2 @austinabell sorry had to update readme badges

**Review by austinabell — APPROVED** [review]
new changes lgtm

**Review by ec2 — APPROVED** [review]

**Pull request merged**